### PR TITLE
Improve R_Recombination Generated Tree

### LINKED ISSUE #353: New R_Recombination rules much too fast?
(Issue drafted by/with @rwest)

After the automatically generated R_Recombination rules in PR #334 
the reaction
`C[C]=CC=C + [H]  <=> CC=CC=C` 
gives this reaction rate
```
! Reaction index: Chemkin #7387; RMG #8750
! Template reaction: R_Recombination
! Flux pairs: C5H7(2097), C5H8(545); H(3)(1), C5H8(545); 
! BM rule fitted to 2 training reactions at node Root_1R->H_N-2R-inRing_N-2R->H_N-2CNOS->S_2CNO->C_Ext-2C-R_Sp-3R!H=2C_Ext-2C-R_Ext-3R!H-R
!     Total Standard Deviation in ln(k): 11.5401827615
! Exact match found for rate rule [Root_1R->H_N-2R-inRing_N-2R->H_N-2CNOS->S_2CNO->C_Ext-2C-R_Sp-3R!H=2C_Ext-2C-R_Ext-3R!H-R]
! Euclidian distance = 0
! family: R_Recombination
H(3)(1)+C5H7(2097)=C5H8(545)                        1.554090e+97 -24.703   0.000    
```
hitting this rate rule:
https://github.com/ReactionMechanismGenerator/RMG-database/pull/334/files#diff-2553ec1602f5d95579cd517d15efcaa5R1734

The [old rate](https://rmg.mit.edu/database/kinetics/reaction/reactant1=multiplicity%202%0A1%20H%20u1%20p0%20c0%0A__reactant2=multiplicity%202%0A1%20%20C%20u0%20p0%20c0%20%7B5,S%7D%20%7B6,S%7D%20%7B7,S%7D%20%7B8,S%7D%0A2%20%20C%20u0%20p0%20c0%20%7B3,S%7D%20%7B4,D%7D%20%7B9,S%7D%0A3%20%20C%20u0%20p0%20c0%20%7B2,S%7D%20%7B5,D%7D%20%7B10,S%7D%0A4%20%20C%20u0%20p0%20c0%20%7B2,D%7D%20%7B11,S%7D%20%7B12,S%7D%0A5%20%20C%20u1%20p0%20c0%20%7B1,S%7D%20%7B3,D%7D%0A6%20%20H%20u0%20p0%20c0%20%7B1,S%7D%0A7%20%20H%20u0%20p0%20c0%20%7B1,S%7D%0A8%20%20H%20u0%20p0%20c0%20%7B1,S%7D%0A9%20%20H%20u0%20p0%20c0%20%7B2,S%7D%0A10%20H%20u0%20p0%20c0%20%7B3,S%7D%0A11%20H%20u0%20p0%20c0%20%7B4,S%7D%0A12%20H%20u0%20p0%20c0%20%7B4,S%7D%0A__product1=1%20%20C%20u0%20p0%20c0%20%7B2,S%7D%20%7B6,S%7D%20%7B7,S%7D%20%7B8,S%7D%0A2%20%20C%20u0%20p0%20c0%20%7B1,S%7D%20%7B3,D%7D%20%7B9,S%7D%0A3%20%20C%20u0%20p0%20c0%20%7B2,D%7D%20%7B4,S%7D%20%7B11,S%7D%0A4%20%20C%20u0%20p0%20c0%20%7B3,S%7D%20%7B5,D%7D%20%7B10,S%7D%0A5%20%20C%20u0%20p0%20c0%20%7B4,D%7D%20%7B12,S%7D%20%7B13,S%7D%0A6%20%20H%20u0%20p0%20c0%20%7B1,S%7D%0A7%20%20H%20u0%20p0%20c0%20%7B1,S%7D%0A8%20%20H%20u0%20p0%20c0%20%7B1,S%7D%0A9%20%20H%20u0%20p0%20c0%20%7B2,S%7D%0A10%20H%20u0%20p0%20c0%20%7B4,S%7D%0A11%20H%20u0%20p0%20c0%20%7B3,S%7D%0A12%20H%20u0%20p0%20c0%20%7B5,S%7D%0A13%20H%20u0%20p0%20c0%20%7B5,S%7D%0A__res=True) is about 10^13 cm3/mol/s.   (10^7 m3/mol/s), about the collision limit.

The new one is (back of the envelope estimate) 10^(97-75)=10^22 cm3/mol/s at 1000K and 10^(97-50)=10^47 cm3/mol/s at 100K.

This is just one example. There are many reactions showing up with rates far exceeding the collision limit, causing stiffness issues.

Any suggestions?

### PR DESCRIPTION
**Pull request by mjohnson541**
Improved generated tree using SABIC_aromatics and generating resonance structures before calculating thermo. Should mostly fix #353.

### CODE REVIEW

File: input/kinetics/families/R_Recombination/rules.py [hunk 1]
```diff
@@ -20,2851 +20,2866 @@
...
-    kinetics = ArrheniusBM(A=(5e+07,'m^3/(mol*s)'), n=-1.21605e-07, w0=(173000,'J/mol'), E0=(17300,'J/mol'), Tmin=(300,'K'), Tmax=(1500,'K'), uncertainty=RateUncertainty(mu=0.0, var=33.1368631905, Tref=1000.0, N=1, correlation='Root_N-1R->H_N-1CClNOSSi->N_N-1COS->O_1CS->C_1C-inRing_Ext-1C-R_Sp-3R!H-1C_2R-inRing',), comment="""BM rule fitted to 2 training reactions at node Root_N-1R->H_N-1CClNOSSi->N_N-1COS->O_1CS->C_1C-inRing_Ext-1C-R_Sp-3R!H-1C_2R-inRing
-    Total Standard Deviation in ln(k): 11.5401827615"""),
+    label = "Root_N-1R->H_N-1CNOS->N_N-1COS->O_1CS->C_N-1C-inRing_Ext-2R-R_Ext-2R-R_Ext-2R-R_Ext-1C-R_6R!H->C",
+    kinetics = ArrheniusBM(A=(1.53609e+09,'m^3/(mol*s)'), n=-0.946459, w0=(173000,'J/mol'), E0=(17300,'J/mol'), Tmin=(300,'K'), Tmax=(1500,'K'), uncertainty=RateUncertainty(mu=0.000724761308015, var=1.22133425516, Tref=1000.0, N=6, correlation='Root_N-1R->H_N-1CNOS->N_N-1COS->O_1CS->C_N-1C-inRing_Ext-2R-R_Ext-2R-R_Ext-2R-R_Ext-1C-R_6R!H->C',), comment="""BM rule fitted to 2 training reactions at node Root_N-1R->H_N-1CNOS->N_N-1COS->O_1CS->C_N-1C-inRing_Ext-2R-R_Ext-2R-R_Ext-2R-R_E
```
Comment (mliu49) [inline input/kinetics/families/R_Recombination/rules.py]: This rate rule seems like it could be problematic.

**Comment by mliu49** [comment]
In RMG-tests, there are a lot of reactions which were estimated by rate rules on master but estimated by exact match to training reactions on this branch. Do you have any idea why that would be?

**Comment by mjohnson541** [comment]
I'll take a look tomorrow, this shouldn't happen and isn't desirable.

**Comment by mjohnson541** [comment]
Should be fixed now

**Pull request merged**